liota/dccs/iotcv2.py

```diff
@@ -0,0 +1,346 @@
+from liota.lib.protocols.helix_protocol import HelixProtocol
+from liota.entities.metrics.metric import Metric
+from liota.lib.utilities.utility import LiotaConfigPath, getUTCmillis, mkdir, read_liota_config
+from liota.lib.utilities.si_unit import parse_unit
+from liota.entities.metrics.registered_metric import RegisteredMetric
+from liota.entities.registered_entity import RegisteredEntity
+
+log = logging.getLogger(__name__)
+
+timeout = 300
+
+class Request:
+    def __init__(self, guid, operation_id, user_queue):
+        self.guid = guid
+        # assume operation_id will be unique in one process
+        self.operation_id = operation_id
+        self.user_queue = user_queue
+
+class Iotcv2(DataCenterComponent):
+    """ The implementation of VMware IoTC V2 Data Center Component
+
+    """
+
+    def __init__(self, con):
+        log.info("Logging into IOTC V2 DCC")
+        self.comms = con
+        if not self.comms.identity.username:
+            log.error("Username not found")
+            raise ValueError("Username not found")
+        elif not self.comms.identity.password:
+            log.error("Password not found")
+            raise ValueError("Password not found")
+        thread = threading.Thread(target=self.comms.receive)
+        thread.daemon = True
+        # This thread will continuously run in background to receive response or actions from DCC
+        thread.start()
+        time.sleep(0.5)
+        self.counter = 0
+
+        self.recv_msg_queue = self.comms.userdata
+        self.req_ops_lock = Lock()
+        self.req_dict = {}
+        thread = threading.Thread(target=self._dispatch_recvd_msg)
+        thread.daemon = True
+        # This thread will continuously run in background to check received responses
+        thread.start()
+
+    def register(self, entity_obj):
+        if isinstance(entity_obj, Metric):
+            # reg_entity_id should be parent's one: not known here yet
+            # will add in creat_relationship();
+            log.info("Registering metric {0} locally".format(entity_obj.name))
+            return RegisteredMetric(entity_obj, self, None)
+        else:
+            log.info("Registering resource with IoTCV2 {0}".format(entity_obj.name))
+            # register fun internal queue
+            reg_resp_q = Queue.Queue()
+
+            def on_response(msg, reg_resp_q):
+                try:
+                    log.debug("Received msg: {0}".format(msg))
+                    json_msg = json.loads(msg)
+                    log.debug("Processing msg: {0}".format(json_msg["type"]))
+                    if json_msg["type"] == "registerMO_rsp" and json_msg["guid"] != "null" and \
+                                    json_msg["guid"] == entity_obj.entity_id:
+                        log.info("Received registration response for: {0}".format(json_msg["guid"]))
+                        if json_msg["result"] == 'SUCCESS':
+                            self.reg_entity_id = json_msg["guid"]
```

Comment:
What happens when it is not SUCCESS?

Reply:
self.reg_entity_id will not be set, then RegistrationFailure() will be raised.